### PR TITLE
collectd: validate config at build time

### PR DESCRIPTION
**Pull request by symphorien**
###### Motivation for this change

validate config at build time.
tested with a newly introduced nixos test.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/monitoring/collectd.nix [hunk 1]
```diff
@@ -43,6 +52,16 @@ in {
   options.services.collectd = with types; {
     enable = mkEnableOption "collectd agent";
 
+    validateConfig = mkOption {
+      default = true;
```
Comment (bjornfor) [inline nixos/modules/services/monitoring/collectd.nix:56]: I wonder if we should do `default = stdenv.hostPlatform == stdenv.buildPlatform;`? (And `defaultText = "stdenv.hostPlatform == stdenv.buildPlatform"`?). (Not important to me, and I don't know how other modules handle this.)

File: nixos/tests/collectd.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          load = "";
+        };
+      };
+      environment.systemPackages = [ pkgs.rrdtool ];
+    };
+
+  testScript = ''
+    machine.wait_for_unit("collectd.service")
+    hostname = machine.succeed("hostname").strip()
+    file=f"/var/lib/collectd/rrd/{hostname}/load/load.rrd"
```
Comment (bjornfor) [inline nixos/tests/collectd.nix:24]: Nit: missing whitespace around `=`in `file=...`.

**Review by bjornfor — APPROVED** [review]
Nit: change commit title "collectd: add nixos test" to "nixos/collectd: add test".

File: nixos/modules/services/monitoring/collectd.nix (hunk 1 above)
Comment (symphorien) [reply nixos/modules/services/monitoring/collectd.nix:56]: I have had inconsistent reviews on similar PRs in the past, so let's leave it as is unless someone has a strong opinion.

**Pull request merged**